aten/src/ATen/native/sparse/SparseTensor.cpp

```diff
@@ -148,79 +176,97 @@ SparseTensor new_with_tensor_and_size_sparse(const LongTensor& indices, const Te
   }
 
   const SparseType& dtype = values.type().toSparse();
+
+#ifndef USE_TH_SIZE_ZERO_DIM
   // NB: This used to be dims, but mumble TH handling zero-sized tensors
   // incorrectly
   if (indices.numel() == 0 && values.numel() == 0) {
     return new_with_size_sparse(dtype, sizes);
   }
+#endif
 
   int64_t sparseDims = indices.size(0);
   int64_t denseDims = values.dim() - 1;
-  AT_CHECK(sizes.size() == sparseDims + denseDims, "number of dimensions must be sparseDims (", sparseDims, ") + denseDims (", denseDims, "), but got ", sizes);
-
-  LongTensor max_indices = std::get</* values */ 0>(indices.max(/* dim */ 1, /* keepdim */ false));
-  LongTensor cpu_max_indices;
-  if (max_indices.is_cuda()) {
-    cpu_max_indices = at::CPU(kLong).copy(max_indices);
-  } else {
-    cpu_max_indices = max_indices;
-  }
-  auto cpu_max_indices_accessor = cpu_max_indices.accessor<int64_t, 1>();
-  for (int64_t d = 0; d < sparseDims; d++) {
-    // NB: This used to sync ndim times to access each entry; now we copy
-    // everything to CPU first and then access it.
-    int64_t max_index_in_dim = cpu_max_indices_accessor[d];
-    int64_t dim_size = sizes[static_cast<size_t>(d)];
-    AT_CHECK(max_index_in_dim < dim_size,
-             "sizes is inconsistent with indices: for dim ", d, ", size is ", dim_size, " but found index ", max_index_in_dim);
+  AT_CHECK(sizes.size() == sparseDims + denseDims, "number of dimensions must be sparseDims (", sparseDims, ") + denseDims (", denseDims, "), but got ", sizes.size());
+
+  // Check to make sure all indices are within the boundaries of `sizes`
+  if (indices.numel() > 0) {
+    LongTensor max_indices = std::get</* values */ 0>(indices.max(/* dim */ 1, /* keepdim */ false));
+    LongTensor cpu_max_indices;
+    if (max_indices.is_cuda()) {
+      cpu_max_indices = at::CPU(kLong).copy(max_indices);
+    } else {
+      cpu_max_indices = max_indices;
+    }
+    auto cpu_max_indices_accessor = cpu_max_indices.accessor<int64_t, 1>();
+    for (int64_t d = 0; d < sparseDims; d++) {
+      // NB: This used to sync ndim times to access each entry; now we copy
+      // everything to CPU first and then access it.
+      int64_t max_index_in_dim = cpu_max_indices_accessor[d];
+      int64_t dim_size = sizes[static_cast<size_t>(d)];
+      AT_CHECK(max_index_in_dim < dim_size,
+               "sizes is inconsistent with indices: for dim ", d, ", size is ", dim_size, " but found index ", max_index_in_dim);
+    }
   }
+
+#ifndef USE_TH_SIZE_ZERO_DIM
```

Comment:
The dense dim invariant is already checked in `set_indices_and_values_unsafe` as part of `_new_with_dims_and_tensor_sparse`. We could keep it here but it will probably be redundant.